shared/actions/chat.js

```diff
@@ -1163,13 +1171,43 @@ function _unboxedToMessage (message: MessageUnboxed, idx: number, yourName, your
     }
   }
 
+  if (message.state === LocalMessageUnboxedState.error) {
+    const error = message.error
+    if (error) {
+      switch (error.errType) {
+        case LocalMessageUnboxedErrorType.misc:
+        case LocalMessageUnboxedErrorType.badversionCritical: // fallthrough
+        case LocalMessageUnboxedErrorType.identify: // fallthrough
+          return {
+            conversationIDKey,
+            key: idx,
+            messageID: error.messageID,
+            reason: error.errMsg || '',
+            timestamp: Date.now(),
+            type: 'Error',
+          }
+        case LocalMessageUnboxedErrorType.badversion:
+          return {
+            conversationIDKey,
+            key: idx,
+            data: message,
+            messageID: idx,
+            timestamp: Date.now(),
+            type: 'InvisibleError',
+          }
+      }
+    }
+  }
+
+  errorIdx++
+
   return {
-    type: 'Error', // TODO
+    type: 'Error',
     messageID: idx,
```

Comment:
We shouldn't be using `idx` for `messageID` here -- this can recur since they're just array indexes as we iterate through received messages in `loadMoreMessages`. We should get rid of this `idx` argument since it's only used here, and it's useless.